docs/configuration.md

```diff
@@ -0,0 +1,91 @@
+# Configuration
+
+### allowUnused
+array of strings or regular expressions to match against a module name
+to determine if it is allowed to be unused
+
+Please create an [issue](https://github.com/charlierudolph/dependency-lint/issues)
+anytime you need to use this
+
+default:
+```yml
+allowUnused: []
```

Comment:
The user will lose the default if they decide to remove the value from the config (or if its a new option and thus not listed in their config). I could just link to the default config from this file.